askama_parser/benches/from_str/main.rs

```diff
@@ -0,0 +1,40 @@
+use askama_parser::{Ast, Syntax};
+use criterion::{black_box, criterion_group, criterion_main, Criterion, Throughput};
+
+criterion_main!(benches);
+criterion_group!(benches, librustdoc);
+
+fn librustdoc(c: &mut Criterion) {
+    let mut group = c.benchmark_group("librustdoc");
+
+    let mut add_benchmark = |name: &str, src: &str| {
+        group.throughput(Throughput::Bytes(src.len() as u64));
+        group.bench_function(name, |b| {
+            let syntax = &Syntax::default();
+            b.iter(|| Ast::from_str(black_box(src), None, black_box(syntax)).unwrap());
+        });
+    };
+
+    let all: String = LIBRUSTDOC.iter().map(|&(_, src)| src).collect();
+    add_benchmark("all", &all);
+
+    for (name, src) in LIBRUSTDOC {
+        add_benchmark(name, src);
+    }
+
+    group.finish();
+}
+
+macro_rules! sources {
+    ($prefix:tt . $suffix:tt => $($name:tt),*) => {
+        &[ $(
+            ($name, include_str!(concat!("./", $prefix, "/", $name, ".", $suffix))),
+        )* ]
+    };
+}
```

Comment:
IMO this macro doesn't carry its weight, let's just expand it in place?

Reply:
Expanded.